src/Translatable/Validation/Rules/TranslatableExists.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Astrotomic\Translatable\Validation\Rules;
+
+use Closure;
+use Illuminate\Contracts\Validation\ValidationRule;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Str;
+
+/**
+ * Custom exists validation for translatable attributes
+ * 
+ * @author Amjad BaniMattar <amjad.banimattar@gmail.com>
+ */
+class TranslatableExists implements ValidationRule
+{
+    /**
+     * The ID that should be ignored.
+     *
+     * @var mixed
+     */
+    protected mixed $ignore = null;
+
+    /**
+     * The name of the ID column.
+     *
+     * @var string
+     */
+    protected string $idColumn = 'id';
+
+    /**
+     * The default locale
+     * 
+     * @var string
+     */
+    protected ?string $locale = null;
+
+    public function __construct(private string $model, private string $field)
```

Comment:
Please `protected` and not `private`. Makes it easier to adjust the rule if needed in userland.
The doc-block should also say `class-string<Model>` and/or it should do a class exists check. As otherwise the `validate` will just explode.

Reply:
On top you could also fill the `$idColumn` from the passed in model by calling `getKeyName()`.